.github/workflows/build.yaml

```diff
@@ -201,37 +205,55 @@ jobs:
 
 
   publishdocker:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     needs: [setup,build,fvtest,src_audit]
     if: startsWith(github.ref, 'refs/tags/')
+
+    permissions:
+      contents: read
+      packages: write
+
     steps:
-      - uses: actions/setup-node@v3
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
         with:
-          node-version: '16.x'
-      - uses: actions/download-artifact@v3
+          buildkitd-flags: --debug
+          config-inline: |
+            [worker.oci]
+              max-parallelism = 1
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Login to the ${{ env.DOCKER_REGISTRY }} Container Registry
+        uses: docker/login-action@v2
         with:
-          name: nodeenv-docker-image
-          path: build/          
-      - name: Push to registry
-        run: |
-          set -xev
-          wget -qO "$PWD/manifest-tool" https://github.com/estesp/manifest-tool/releases/download/v1.0.0/manifest-tool-linux-amd64
-          chmod +x ./manifest-tool
+          registry: ${{ env.DOCKER_REGISTRY }}
+          username: ${{ env.DOCKER_REGISTRY == 'docker.io' && secrets.DOCKERHUB_USERNAME || github.actor }}
+          password: ${{ env.DOCKER_REGISTRY == 'docker.io' && secrets.DOCKERHUB_TOKEN    || secrets.GITHUB_TOKEN }}
 
-          echo "Version strings are ${PACKAGE_VERSION} ${MINOR_PACKAGE_VERSION}"
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.DOCKER_REGISTRY }}/${{ github.repository_owner }}/fabric-nodeenv
+          tags: |
+            type=semver,pattern={{version}}
```

Comment:
Where are these versions being defined... can't see them...

Reply:
how stupid of me to not spot that :-)

Reply:
It only took me about 2-3 days of staring at the syntax to start to piece this together!

The GHA metadata logic seems like it is generally _doing the right thing_ with respect to applying image tags based on semver conventions.  I had a chance to try out the type=[ref](https://github.com/docker/metadata-action#typeref) syntax, which looks _great_.  This means we could use the semrev conventions in the branch names to infer the intent of the tag.